### DISCUSSION TITLE
Not able to attach to a sub process

### DISCUSSION THREAD

**Discussion by JeevananthamSathish** [microsoft/debugpy#1698] [Q&A]
I have a created a c# exe app where i used pythonnet and running a python script inside the dotnet app as a sub-process . I am trying to debug the python script which is running inside the donet app but in vscode  i am not able to achieve this. I tried remote debugging with connect listen logic ... kindly help me with debugging the python subprocess script ...

**Comment by rchiodo** [comment]
Can you share some example code? I can try it myself.

Reading about [Pythonnet](https://pythonnet.github.io/) it sounds like it should work, but there might be something that pythonnet is doing that messes up attach.

There's also a `logToFile` setting you can set in VS code's launch.json. This will output a bunch of log files in your <extensions folder>\ms-python.debugpy-<version> folder.

**Comment by JeevananthamSathish** [comment]
HI Rich Chiodo,

  *
Thanks for taking time to respond my query
  *
I actually found the issue and resolved it.
     *
I had path mappings which was causing the issue i removed it - then it worked like a magic
  *
But i had to call configure method before calling listen - in python script otherwise not attaching.

Python Script
debugpy.configure(subprocess=False)
    debugpy.listen(5678)
    debugpy.wait_for_client()  # This will pause the execution until a debugger attaches
    debugpy.breakpoint()

config json:
 "configurations": [
        {
            "name": "Python Debugger: Remote Attach",
            "type": "debugpy",
            "request": "attach",
            "connect": {
                "host": "localhost",
                "port": 5678
            },
            "justMyCode": false
        }
    ]
My Pythonnet c# code :
      PythonEngine.PythonHome = @"C:\Users\SATHISHKUMAR\AppData\Local\Programs\Python\Python39";

 // Add the Python standard library to Python path
 PythonEngine.PythonPath = Environment.GetEnvironmentVariable("PYTHONPATH") + ";" +
                           @"C:\Users\SATHISHKUMAR\AppData\Local\Programs\Python\Python39\Lib";
 PythonEngine.DebugGIL = true;

 PythonEngine.Initialize();


 try
 {

     string pythonScriptPath = "D:\\debugTestPython\\script.py";
     if (File.Exists(pythonScriptPath))
     {
         string script = File.ReadAllText(pythonScriptPath);

         using (Py.GIL()) // Ensures proper handling of Python Global Interpreter Lock (GIL)
         {
             // Execute the Python code

             PythonEngine.Exec(script);

         }

     }
     else
     {
         Console.WriteLine($"File {pythonScriptPath} not found.");
     }
     //PythonEngine.Shutdown();
 }

  *
Now i can able to attach but whenever i inspect a variable the tool crashes. Any solution.?
  *
Anything wrong in my configuration or script - kindly share your suggestions to improve.

Regards
Sathishkumar Jeevanantham
________________________________

**Reply by rchiodo** [reply]
VS code crashes when inspect a variable? You might try adding this setting:

```json
"logToFile": true
```

Or using the PYDEVD_DEBUG_FILE environment variable. It should be set to a directory that exists.

**Reply by rchiodo** [reply]
That will generate a log file that hopefully has the crash for the debugger in it. My guess is something that pythonnet does prevents evaling different variables

**Comment by JeevananthamSathish** [comment]
Forget to mention :

  *
Whenever the debugg adapter attaches it opened a new tab in vscode with title <string> :
  *
To solve this -> i added a environment variable PATHS_FROM_ECLIPSE_TO_PYTHON to map to my python script path against this <script > like this [["D:/debugTestPython/script.py","<string>"]].
  *
Then it worked ,it opens the correct python script as mentioned in the env path but it opens a new tab with same script file in vscode.

Any solution to solve this problem.?


Thanks in advance

Regards
Sathishkumar
________________________________

**Reply by rchiodo** [reply]
Sorry but that's by design. The pythonnet is running your python code as a string. As far as VS code and debugpy know, that's a different file.

**Reply by rchiodo** [reply]
Maybe pythonnet can run a file instead?
